federation/pkg/federation-controller/replicaset/planner/planer.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package planer
+
+import (
+	"sort"
+
+	fed_api "k8s.io/kubernetes/federation/apis/federation"
+)
+
+// Planer decides how many out of the given replicas should be placed in each of the
+// federated clusters.
+type Planer struct {
+	preferences *fed_api.FederatedReplicaSetPreferences
+}
+
+type namedLocalReplicaSetPreferences struct {
+	clusterName string
+	fed_api.LocalReplicaSetPreferences
+}
+
+type byWeight []*namedLocalReplicaSetPreferences
```

Comment:
It is for sorting. See first example here https://golang.org/pkg/sort/. Unfortunately golang is absolutely disgusting when it comes to generic functions and interfaces.